test/nodes/test_prompt_node.py

```diff
@@ -344,6 +344,21 @@ def test_complex_pipeline(prompt_model):
     assert "berlin" in result["results"][0].casefold()
 
 
+@pytest.mark.integration
+@pytest.mark.parametrize("prompt_model", ["hf", "openai"], indirect=True)
+def test_simple_pipeline_with_topk(prompt_model):
+    if prompt_model.api_key is not None and not is_openai_api_key_set(prompt_model.api_key):
+        pytest.skip("No API key found for OpenAI, skipping test")
+
+    node = PromptNode(prompt_model, default_prompt_template="question-generation", top_k=2)
+
+    pipe = Pipeline()
+    pipe.add_node(component=node, name="prompt_node", inputs=["Query"])
+    result = pipe.run(query="not relevant", documents=[Document("Berlin is the capital of Germany")])
```

Comment:
Thanks for adding the test! I wanted to ask if it would make more sense to be setting `query=None` instead of `"not relevant"` for general use case when using a node that ignores the query? Doesn't matter for tests though, just wanted to ask.

Reply:
@sjrl I think there is still something that required the query to be set. Ideally, we shouldn't have to add `query` add all here. As long `query` is required, it doesn't make a difference, I'd say.